lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -252,6 +254,35 @@ data TransactionLayer k tx = TransactionLayer
         --
         -- Returns `Nothing` for ByronEra transactions.
 
+    , distributeSurplus
+        :: FeePolicy
+        -> Coin -- Surplus to distribute
+        -> TxFeeAndChange -- Fee and value of relevant change output (if any)
+        -> Either Coin TxFeeAndChange
```

Comment:
Should write something about the `Left Coin`. 

On the call-site in `balanceTransaction` it should never happen, because we make sure to overestimate the fee-requirement for the initial coin-selection.